recipes/rmlui/4.x/conanfile.py

```diff
@@ -85,78 +86,96 @@ def requirements(self):
+            "if(LUA_FOUND)": "if(TRUE)", # only in 4.0
+            "list(include_directories(${LUA_INCLUDE_DIR}))": "", # only in 4.0
+            "list(APPEND LUA_BINDINGS_INCLUDE_DIRS ${LUA_INCLUDE_DIR})": "",
+            "list(APPEND LUA_BINDINGS_LINK_LIBS ${LUA_LIBRARIES})": "list(APPEND LUA_BINDINGS_LINK_LIBS lua::lua)",
             # disables the built-in generation of package configuration files
             "if(PkgHelpers_AVAILABLE)": "if(FALSE)"
         }
 
+        if self.version >= "4.4":
+            inject_robinhood_dep = """
+find_package(robin_hood REQUIRED)
+target_link_libraries(RmlCore PUBLIC robin_hood::robin_hood)
+if( CUSTOM_CONFIGURATION AND CUSTOM_LINK_LIBRARIES )
+"""
+        else:
+            inject_robinhood_dep = """
+find_package(robin_hood REQUIRED)
+target_link_libraries(RmlCore robin_hood::robin_hood)
+if( CUSTOM_CONFIGURATION AND CUSTOM_LINK_LIBRARIES )
+"""
+        if self.options.with_thirdparty_containers:
+            replace_mapping['if( CUSTOM_CONFIGURATION AND CUSTOM_LINK_LIBRARIES )'] = inject_robinhood_dep
+
         cmakelists_path = os.path.join(
-            self._source_subfolder, "CMakeLists.txt")
+            self.source_folder, "CMakeLists.txt")
         for key, value in replace_mapping.items():
-            tools.replace_in_file(cmakelists_path, key, value, strict=False)
+            replace_in_file(self, cmakelists_path, key, value, strict=False)
 
         if self.options.with_thirdparty_containers:
-            config_path = os.path.join(self._source_subfolder,
+            config_path = os.path.join(self.source_folder,
                                        "Include", "RmlUi", "Config", "Config.h")
-            tools.replace_in_file(
-                config_path, "\"../Core/Containers/robin_hood.h\"", "<robin_hood.h>")
+            replace_in_file(
+                self, config_path, "\"../Core/Containers/robin_hood.h\"", "<robin_hood.h>")
 
     def build(self):
         self._patch_sources()
-        self._configure_cmake().build()
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
 
     def package(self):
-        self._configure_cmake().install()
-        self.copy("*LICENSE.txt", dst="licenses", src=self._source_subfolder, excludes=("Samples/*", "Tests/*"))
+        copy(self, pattern="*LICENSE.txt", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder, excludes=("Samples/*", "Tests/*"))
+        cmake = CMake(self)
+        cmake.install()
 
     def package_info(self):
         if self.options.matrix_mode == "row_major":
             self.cpp_info.defines.append("RMLUI_MATRIX_ROW_MAJOR")
 
         if not self.options.enable_rtti_and_exceptions:
             self.cpp_info.defines.append("RMLUI_USE_CUSTOM_RTTI")
+            if self.settings.compiler == "gcc" or self.settings.compiler == "clang":
+                self.cpp_info.cxxflags.append("-fno-rtti -fno-exceptions")
```

Comment:
This should be controlled from the end user not the recipe 🤔 this wont be pushed to other parts of the graph which I am sure will break too

Reply:
The option was in the original recipe but was broken. Would it be better to just remove the option entirely?

Reply:
I decided to just remove the option entirely. I'm not sure it ever worked.

Reply:
When I originally created the recipe I think I simply added an option for every CMake option that made sense (reference: https://mikke89.github.io/RmlUiDoc/pages/cpp_manual/building_with_cmake.html). It comes into play around https://github.com/mikke89/RmlUi/blob/master/CMakeLists.txt#L575, so I guess it does achieve its intended effect, if only for this particular library and not even its direct dependencies.

I personally think it wouldn't hurt to keep the option as it was, although it could certainly be argued that it wouldn't see much use anyway.

Reply:
Ok, hmmm. I think that the issues that @prince-chrismc brought up mean the option could cause breakages. Maybe lets leave it off for now and we can re-add if there are people who need it?

Reply:
Yeah, that's perfectly fine.

Reply:
It's generally not a good idea to remove options since that will break consumers. 🤔 

I think this is fine, if you want to control RTTI it's really a graph level thing and you should have Conan inject those flags globally, setting it here and let the project do it sure but it's not enough

and force flags downstream is not a good idea, it's very difficult to track down